:100644 100644 2637a57... 342d8f5... M	src/usr/local/www/diag_dump_states.php
:100644 100644 525bc88... 796a32b... M	src/usr/local/www/system_advanced_misc.php

diff --git a/src/usr/local/www/diag_dump_states.php b/src/usr/local/www/diag_dump_states.php
index 2637a57..342d8f5 100644
--- a/src/usr/local/www/diag_dump_states.php
+++ b/src/usr/local/www/diag_dump_states.php
@@ -205,14 +205,10 @@ print $form;
 		$arr[] = array("filter" => $_POST['filter']);
 	}
 
-	if (isset($_POST['filter']) || !isset($config['system']['require_state_filter'])) {
-		if (count($arr) > 0) {
-			$res = pfSense_get_pf_states($arr);
-		} else {
-			$res = pfSense_get_pf_states();
-		}
+	if (count($arr) > 0) {
+		$res = pfSense_get_pf_states($arr);
 	} else {
-		$res = NULL;
+		$res = pfSense_get_pf_states();
 	}
 
 	$states = 0;
@@ -265,9 +261,6 @@ print $form;
 if ($states == 0) {
 	if (isset($_POST['filter']) && !empty($_POST['filter'])) {
 		$errmsg = gettext('No states were found that match the current filter.');
-	} else if (!isset($_POST['filter']) && isset($config['system']['require_state_filter'])) {
- 		$errmsg = gettext('State display suppressed without filter submission. '.
- 		'See System > Advanced, Miscellaneous, Require State Filter.');
 	} else {
 		$errmsg = gettext('No states were found.');
 	}
diff --git a/src/usr/local/www/system_advanced_misc.php b/src/usr/local/www/system_advanced_misc.php
index 525bc88..796a32b 100644
--- a/src/usr/local/www/system_advanced_misc.php
+++ b/src/usr/local/www/system_advanced_misc.php
@@ -50,7 +50,6 @@ $pconfig['powerd_enable'] = isset($config['system']['powerd_enable']);
 $pconfig['crypto_hardware'] = $config['system']['crypto_hardware'];
 $pconfig['thermal_hardware'] = $config['system']['thermal_hardware'];
 $pconfig['schedule_states'] = isset($config['system']['schedule_states']);
-$pconfig['require_state_filter'] = isset($config['system']['require_state_filter']);
 $pconfig['gw_down_kill_states'] = isset($config['system']['gw_down_kill_states']);
 $pconfig['skip_rules_gw_down'] = isset($config['system']['skip_rules_gw_down']);
 $pconfig['use_mfs_tmpvar'] = isset($config['system']['use_mfs_tmpvar']);
@@ -217,12 +216,6 @@ if ($_POST) {
 			unset($config['system']['schedule_states']);
 		}
 
- 		if ($_POST['require_state_filter'] == "yes") {
- 			$config['system']['require_state_filter'] = true;
- 		} else {
- 			unset($config['system']['require_state_filter']);
- 		}
-
 		if ($_POST['gw_down_kill_states'] == "yes") {
 			$config['system']['gw_down_kill_states'] = true;
 		} else {
@@ -477,17 +470,6 @@ $section->addInput(new Form_Checkbox(
 	'schedule are killed. This option overrides that behavior by not clearing states '.
 	'for existing connections.');
 
-$section = new Form_Section('Large Data Set Behavior');
- 
-$section->addInput(new Form_Checkbox(
- 	'require_state_filter',
- 	'Require State Filter',
- 	'Do not display state table without a filter',
-	$pconfig['require_state_filter']
-))->setHelp('By default, the entire state table is displayed when entering '.
-	'Diagnostics > States. This option requires a filter to be entered '.
-	'before the states are displayed. Useful for systems with large state tables.');
-
 $form->add($section);
 $section = new Form_Section('Gateway Monitoring');
 
